docker.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+set -x
+RUST_TOOLCHAIN=1.49
+# TODO: check if it's defined, otherwise use default:
+ARCH=x86_64
+GIT_COMMIT=$(git rev-parse HEAD)
+GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+DOCKER_TAG=rustvmm/dev
+
+# Get docker tags
+dtags () {
+    local image="${1}";
+    wget -c -q https://registry.hub.docker.com/v1/repositories/"${image}"/tags -O - | tr -d '[]" ' | tr '}' '\n' | awk -F: '{print $3}'
+}
+
+# Get the latest published version. Returns a number.
+# If latest is v100, returns 100.
+latest(){
+  dtags $DOCKER_TAG | grep -v "_" | cut -c 2- | sort -n | tail -1
+}
+
+build(){
+  new_version=$(($1 + 1))
+  new_tag=${DOCKER_TAG}:${new_version}_$ARCH
+  echo "Bulding version new_version"
+  #set -e
+  echo docker build -t $new_tag \
+        --build-arg RUST_TOOLCHAIN=${RUST_TOOLCHAIN} \
+        --build-arg GIT_BRANCH=${GIT_BRANCH} \
+        --build-arg GIT_COMMIT=${GIT_COMMIT} \
+        -f Dockerfile .
+  #set +e
+  # Add a tag alias for pointing latest to the new_tag
+  docker tag $new_tag ${DOCKER_TAG}:latest
+  echo "Build completed for $new_tag"
+}
+
+publish(){
+    new_version=$(($1 + 1))
+    new_tag=${DOCKER_TAG}:${new_version}_$ARCH
+    echo "Publishing $new_version and 'latest' on dockerhub"
```

Comment:
We shouldn't publish this one as latest because the latest tag should be used on the manifest (i.e. the image that links to either x86_64 or aarch64 depending on the platform from which pull is called).